### PR TITLE
Add filtering on the workflow id for active states endpoint

### PR DESCRIPTION
**Pull request by tnsetting**
# Hey, I just made a Pull Request!

## Checklist for PR author(s)

- [x] Changes are covered by unit test
- [ ] All tests pass
- [ ] Code coverage check passes
- [ ] Error handling is tested
- [ ] Errors are handled at the appropriate layer
- [ ] Errors that cannot be handled where they occur are propagated
- [ ] (optional) Changes are covered by system test
- [ ] Relevant documentation updated
- [x] This PR has NO breaking change to public API
- [ ] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)

- [x] This PR has been incorporated in release note for the coming version
- [ ] Risky changes introduced by this PR have been all considered

### CODE REVIEW

File: styx-api-service/bin/run-service.sh [hunk 1]
```diff
@@ -1,3 +1,3 @@
 #!/usr/bin/env bash
 
-exec java -jar styx-api-service/target/styx-api-service.jar "$@"
+exec java -jar styx-api-service/target/styx-api-service-2.1.46-SNAPSHOT.jar "$@"
```
Comment (NatashaL) [inline styx-api-service/bin/run-service.sh]: Let's revert this change

File: styx-api-service/src/test/java/com/spotify/styx/api/StatusResourceTest.java [hunk 2]
```diff
@@ -82,6 +87,9 @@
   private static final RunState OTHER_PERSISTENT_STATE = RunState.create(OTHER_WFI, State.RUNNING,
       StateData.zero(), Instant.now(), 84L);
 
+  private static final RunState THIRD_PERSISTENT_STATE = RunState.create(OTHER_WFI, State.RUNNING,
```
Comment (NatashaL) [inline styx-api-service/src/test/java/com/spotify/styx/api/StatusResourceTest.java]: This state was intended to refer to the `THIRD_WFI` right?

File: styx-service-common/src/main/java/com/spotify/styx/storage/InMemStorage.java [hunk 3]
```diff
@@ -347,6 +347,15 @@ public void deleteActiveState(WorkflowInstance workflowInstance) {
         .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
   }
 
+  @Override
+  public Map<WorkflowInstance, RunState> readActiveStates(String componentId, String workflowId)
```
Comment (NatashaL) [inline styx-service-common/src/main/java/com/spotify/styx/storage/InMemStorage.java]: Let's add a test for this. Hopefully it will satisfy codecov

Comment (NatashaL) [reply styx-service-common/src/main/java/com/spotify/styx/storage/InMemStorage.java]: As well as tests for DatastoreStorage and AggregateStorage.

File: styx-service-common/src/test/java/com/spotify/styx/storage/DatastoreStorageTest.java [hunk 4]
```diff
@@ -461,6 +461,19 @@ public void shouldReturnAllActiveStatesForAComponent() throws Exception {
     assertThat(activeStates, is(Map.of(WORKFLOW_INSTANCE2, RUN_STATE2)));
   }
 
+  @Test
+  public void shouldReturnActiveStatesForAWORKFLOW() throws Exception {
```
Comment (NatashaL) [inline styx-service-common/src/test/java/com/spotify/styx/storage/DatastoreStorageTest.java]: Caps look scary..

Comment (tnsetting) [reply styx-service-common/src/test/java/com/spotify/styx/storage/DatastoreStorageTest.java]: Indeed. Friday's last commit ;)
